climada/util/coordinates.py

```diff
@@ -1821,6 +1821,160 @@ def apply_box(df_exp):
     }
     return raster_out, meta
 
+def reproject_raster_data(source, src_crs, src_transform, dst_crs=None, dst_transform=None,
+                      dst_shape=None, dst_resolution=None, dst_global_origin=None,
+                      dst_global_extent=None,
+                      dst_buffer=0, resampling=None, conserve=None,
+                      **kwargs):
+    """
+    Convenience wrapper function around rasterio.warp.reproject :
+    Reproject (up- or downsample) 2D np.ndarray to a grid using rasterio.
+    The destination (target) grid is defined from dst_transform and optionally
+    dst_resolution, dst_global_origin, and dst_crs.
+    This function ensures that reprojected data with the same dst_resolution and
+    dst_global_origins are on the same global grid, i.e., no offset between
+    destination grid points for different source grids that are projected to
+    the same target resolution.
+
+    Parameters
+    ----------
+    source : np.ndarray
+        2D-array containing the values to be reprojected
+    src_crs : CRS
+        Source CRS
+    src_transform : Affine
+        Source transform, contains info on grid, e.g.:
+            Affine(0.00833333333333333, 0.0, -18.175000000000068,
+                                 0.0, -0.00833333333333333, 43.79999999999993)
+    dst_crs : CRS, optional
+        Destination CRS. The default is src_crs.
+    dst_transform : Affine, optional
+        Destination transform. The default is src_transform.
+    dst_shape : tuple with integers, optional
+        (height, width) of destination grid. The default is source.shape.
+        Note: Destination shape is changed if resolution changes
+    dst_resolution : number, optional
+        Destination resolution.
+        Overwrites resolution provided in dst_transform.
+        Unit of resolution depends on CRS;
+        Typically but not necessarily, the unit is degree lat, lon.
+        The default is dst_transform[0].
+    dst_global_origin : tuple of numbers, optional
+        global lon and lat origins as basis for destination grid.
+        Only required if dst_resolution different than dst_transform[0].
+        The default is (-180.0, 90), change dst_global_origin for different units.
```

Comment:
I suggest to raise an exception if the `dst_global_origin` parameter is needed, but the user didn't specify any, and the destination CRS is not in latitude/longitude. Maybe even restrict the default value to some known systems like `epsg:4326`.